src/pybind/mgr/volumes/fs/operations/group.py

```diff
@@ -22,6 +22,8 @@ class Group(GroupTemplate):
     def __init__(self, fs, vol_spec, groupname):
         if groupname == Group.NO_GROUP_NAME:
             raise VolumeException(-errno.EPERM, "Operation not permitted for group '{0}' as it is an internal group.".format(groupname))
+        if groupname in vol_spec.INTERNAL_DIRS:
```

Comment:
For `_nogroup` we return `EPERM` because it is valid group but as it is internal to ceph, operations are not allowed.
For `_index`, `_deleting`, `_legacy` we return `EINVAL` because it is not a valid group name.